### PR TITLE
Allow the OOB authentication flow when we can't open a browser.

### PR DESCRIPTION
**Pull request by dlorenc**
I've tested that this works with dex.

Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

File: pkg/oauthflow/interactive.go [hunk 1]
```diff
@@ -53,16 +57,22 @@ func (i *InteractiveIDTokenGetter) GetIDToken(p *oidc.Provider, cfg oauth2.Confi
 		return nil, err
 	}
 
-	authCodeURL := cfg.AuthCodeURL(stateToken, append(pkce.AuthURLOpts(), oauth2.AccessTypeOnline, oidc.Nonce(nonce))...)
-	fmt.Fprintf(os.Stderr, "Your browser will now be opened to:\n%s\n", authCodeURL)
-	if err := open.Run(authCodeURL); err != nil {
+	opts := append(pkce.AuthURLOpts(), oauth2.AccessTypeOnline, oidc.Nonce(nonce))
+	authCodeURL := cfg.AuthCodeURL(stateToken, opts...)
+	var code string
+	if err := open.Run(authCodeURL); err == nil {
```
Comment (letmaik) [inline pkg/oauthflow/interactive.go:63]:
```suggestion
	if err := open.Run(authCodeURL); err != nil {
```

Comment (dlorenc) [reply pkg/oauthflow/interactive.go:63]: Nice catch! Left over from my final testing to simulate a failure :)

**Review by bobcallaway — COMMENTED** [review]
one question otherwise LGTM

File: pkg/oauthflow/interactive.go [hunk 2]
```diff
@@ -97,6 +105,15 @@ func (i *InteractiveIDTokenGetter) GetIDToken(p *oidc.Provider, cfg oauth2.Confi
 	return &returnToken, nil
 }
 
+func doOobFlow(cfg oauth2.Config, authURL string) string {
```
[the remark this answers is not in the document]
Comment (dlorenc) [reply pkg/oauthflow/interactive.go]: Fixed!

**Review by bobcallaway — APPROVED** [review]

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**